### ISSUE TITLE
[psqldef] 「::(cast)」 statement is converted to 「collate」statement in second migration

### ISSUE THREAD

**Issue by you8-hz** [k0kubun/sqldef#278]
First migration is successful but second migration failed.
「::(cast)」 statement is converted to 「collate」statement in second migration.
Please confirm this error.

```shell
$ psqldef --version
v0.13.7
```

## --export output

```sql
CREATE TABLE "public"."hoge" (
    "amount" text
);

CREATE VIEW public.hoge_view AS SELECT (hoge.amount)::numeric(10,2) AS amount_num FROM hoge;
```

## Input SQL

```sql
create table hoge (
    amount   text
);

create view hoge_view as
select
    amount::numeric(10,2) as amount_num
from hoge;
```

## Current output

```sql
CREATE OR REPLACE VIEW "public"."hoge_view" AS select amount collate  as amount_num from hoge;
2022/08/26 19:26:57 pq: syntax error at or near "as"
```

## Expected output

No errors.

**Comment by k0kubun** [comment]
Thanks for reporting it. Now that we have the original PostgreSQL parser https://github.com/k0kubun/sqldef/pull/241, which is currently just partially applied, I think this one is much easier to fix than before. However, as I need to focus on conference preparations for a few weeks, please allow me to have a look at this after Sep 10th or so. I could also look at a pull request if filed.

**Comment by k0kubun** [comment]
Fixed in v0.13.10.
